tools/llvm-cbe/llvm-cbe.cpp

```diff
@@ -398,6 +398,7 @@ static int compileModule(char **argv, LLVMContext &Context) {
   if (RelaxAll) {
     if (FileType != CodeGenFileType::CGFT_ObjectFile)
 #else
+  {
```

Comment:
This may fix the build, but it's not checking for `RelaxAll` any more, which means it will be warning about `-mc-relax-all` regardless of whether it is set?

Can you see if it compiles if you have the same `if (RelaxAll) {` for LLVM 9 and below? Alternatively can you make it skip this warning entirely?

I don't really know what this code is doing though. I am surprised object files are mentioned here at all.

Reply:
Yes, I noticed that warning, but thought it was something on my side, not LLVM and/or llvm-cbe.
I've checked LLVM-6, 7, 8, 9, 10, and 11 (every single version that Ubuntu 20.04 has), and figured out that `RelaxAll` is supported in 8 and 9 too, and using LLVM 6 or 7 does not allow `llvm-cbe` to build successfully, so I added a check for LLVM < 8 into `CMakeLists.txt` too, and used `RelaxAll` for every other version of LLVM < 10.

Checked `unittests/CWriterTest` and `test/selectionsort`, everything is performing normally on LLVM 8 to 11.

Also tried to build `lzlib` and `plzip` using `llvm-cbe`, `lzlib` builds and works perfectly (with any version of LLVM from 8 to 11), but I have no success on `plzip`: sometimes `llvm-cbe` just crashes, sometimes it produces types that have not been declared at all or variables that declared after being used. I'm gonna file a couple issues about that.

Reply:
Thanks, this looks good to me.